django_project/frontend/src/containers/MainPage/Metrics/index.tsx

```diff
@@ -60,6 +63,20 @@ const Metrics = () => {
     const [hasEmptyPropertyAvailable, setHasEmptyPropertyAvailable] = useState(true);
     const [hasEmptyAreaAvailable, setHasEmptyAreaAvailable] = useState(true);
 
+    const { data: userInfoData, isLoading, isSuccess } = useGetUserInfoQuery();
+    const [userRoles, setUserRoles] = useState([]);
+    const [userPermissions, setUserPermissions] = useState([]);
+
+    useEffect(() => {
+        if (isSuccess) {
+            // Preprocess user roles to make them lowercase and remove spaces
+            setUserRoles(userInfoData?.user_roles.map((role) => role.toLowerCase().replace(/\s/g, '')) || []);
```

Comment:
Do we still need this? Can we remove it?

Reply:
@dimasciput we need it ...when I get the user permissions they come as strings so I have to compare/read those strings to determine what permission it is and set true to the appropriate variable that will be checked before rendering the chart

Reply:
@tinashechiraya, this is a user role, not user permissions. Can you direct me to the line where you need to check the user role in this file?

Reply:
@dimasciput from line 239

updateConstants function iterates over the provided userRoles and userPermissions and, for each combination, it calls userRolesHavePermission to update the constants object based on the permissions.

// update constants based on user roles and permissions
    function updateConstants(
        userRoles: string[],
        userPermissions: string[],
        constants: Constants
    ) {
        for (const role of userRoles) {
            for (const permission of userPermissions) {
                userRolesHavePermission(permission)
            }
        }
    }

Reply:
@tinashechiraya yes, but why do you need to loop the userRoles? The data is not used inside the loop?

Reply:
@dimasciput  You are right I see what you mean ,apologies. I have patched it , I removed the user roles since they're not necessary as we are directly checking on the user permissions to determine which chart to render